src/main.py

```diff
@@ -108,31 +51,14 @@ def main():
     print(table)
     print("\n")
 
-    while 1:
-        if keyboard.is_pressed(config.general['resume_key']):
-            screen = Screen(config.general["monitor"])
-            # Run health monitor thread
-            health_manager = HealthManager(screen)
-            health_monitor_thread = threading.Thread(target=health_manager.start_monitor)
-            health_monitor_thread.daemon = True
-            health_monitor_thread.start()
-            # Run death monitor thread
-            death_manager = DeathManager(screen)
-            death_monitor_thread = threading.Thread(target=death_manager.start_monitor)
-            death_monitor_thread.daemon = True
-            death_monitor_thread.start()
-            # Create other "global" instances
-            game_recovery = GameRecovery(screen, death_manager)
-            game_stats = GameStats()
-            run_bot(config, screen, game_recovery, game_stats, death_manager, health_manager)
-            break
-        if keyboard.is_pressed(config.general['auto_settings_key']):
-            adjust_settings()
-            break
-        elif keyboard.is_pressed(config.general['graphic_debugger_key']):
-            run_graphic_debugger()
-            break
-        time.sleep(0.02)
+    keyboard.add_hotkey(config.general['auto_settings_key'], lambda: adjust_settings(config))
+    keyboard.add_hotkey(config.general['graphic_debugger_key'], lambda: run_graphic_debugger())
```

Comment:
I can not exit the graphic debugger anymore with f12. Not sure how to resolve this. Seems like cv2.waitKey() is blocking keyboard? Can you check if you have the same behaviour and look into this?

Reply:
ahhh right, it is because the run_graphic_debugger is not a strictly self-contained function, that means it does its own loop (while 1:...) inside it so on the invokation is not return the keyboard control to the main-loop.

I'll look into this, a solution would be to have the graphic debugger on a separate thread with its own loop (following the same pattern I used for splitting the game controller logic). That would also allow to stop the graphic debugger without killing botty.

Reply:
reading the other comment I now understand better the initial logic behind the initial code, basically you want the following:

run EITHER botty OR graphic_debugger.
Specifically prevent the situation in which you might have both running at the same time.

is that correct?

Reply:
I made some more changes to make the wanted behavior work , id est:

F11 will run botty if  botty is not running and the graphic debugger is not running
F11 will kill the graphic debugger and run botty if the graphic debugger was running
F11 will pause botty if botty was already running

F10 will start the debugger if the grapchi debugger and botty are not running
F10 will stop the graphic debugger if it was running
F10 will kill botty and start the graphic debugger if botty was running